tests/test_compile.py

```diff
@@ -46,6 +49,41 @@ def test_compile_not_enough_args(self):
                 main()
         self.assertEqual(cm.exception.code, 1)
 
+    def test_compile_not_matching_targets(self):
+        stdout_redirect = io.StringIO()
+        sys.stdout = stdout_redirect
+        with self.assertLogs(logger='kapitan.targets', level='ERROR') as cm:
+            # as of now, we cannot capture stdout with contextlib.redirect_stdout
+            # since we only do logger.error(e) in targets.py before exiting
+            with self.assertRaises(SystemExit) as ca:
+                unmatched_filename = "inventory/targets/minikube-es-fake.yml"
+                correct_filename = "inventory/targets/minikube-es.yml"
+                os.rename(src=correct_filename, dst=unmatched_filename)
+                sys.argv = ["kapitan", "compile"]
+
+                try:
+                    main()
+                finally:
+                    # correct the filename again, even if assertion fails
+                    if os.path.exists(unmatched_filename):
+                        os.rename(src=unmatched_filename, dst=correct_filename)
+        error_message = "Target \"minikube-es\" is missing the corresponding yml file"
+        self.assertTrue(' '.join(cm.output).find(error_message) != -1)
```

Comment:
Maybe it's more future proof to check if the output contains an "InventoryError" and "is missing the corresponding yml file" so we avoid hardcoding more names in tests and have to manually change them in the future (?). The rest looks good. :)

Reply:
@uberspot Thanks for the review! 
I removed the target name from the string like you said to make it more generic. As for "InventoryError", it is not actually logged, and we don't seem to be able to capture the error thrown in stdout/error during unittest, so I will pass on it.